controller_interface/include/semantic_components/semantic_component_interface.hpp

```diff
@@ -27,14 +27,19 @@ template <typename MessageReturnType>
 class SemanticComponentInterface
 {
 public:
-  explicit SemanticComponentInterface(const std::string & name, size_t size = 0)
+  SemanticComponentInterface(const std::string & name, std::vector<std::string> interface_names)
+  : name_{name}, interface_names_{interface_names}
+  {
+    state_interfaces_.reserve(interface_names.size());
+  }
+
+  explicit SemanticComponentInterface(const std::string & name, size_t size = 0) : name_{name}
```

Comment:
```suggestion
  explicit SemanticComponentInterface(const std::string & name, std::size_t size = 0u) : name_{name}
```

I think the `0u` is needed here to avoid a warning, no?
otherwise I think you used std::size_t in other places so it felt appropriate to change this

Not sure how consistent we are with this in our codebase though so it's not really a strong comment...

Reply:
Using `0u` is not needed in this case, there is no implicit casting. 

I think in all cases, `std::size_t` should be used since it's C++'s equivalent of size_t (which is c-style define).